### PR TITLE
chore(sdk): Update mlflow version in sdk

### PR DESCRIPTION
**Pull request by deadlycoconuts**
# Description
This PR is the **first part** (1 out of 3) of a series of PRs to update the version of Mlflow used by Merlin (to `1.26.1`*). These changes are:

1. Update the version of Mlflow used in the Merlin SDK and publish it to PyPI
2. Update 
   - the Merlin pyfunc server and batch predictor to use the updated Merlin SDK version released in step 1 and publish them to PyPI
   - the Merlin API server to ensure that requests sent to Mlflow reflect the update API endpoint contracts
   - the Mlflow image so that it's built using version `1.26.1`
3. Update the default pyfunc server and batch predictor version in the Merlin SDK and publish its new version to PyPI

I also made some minor refactoring to the requirements files so that they are stored in a separate file and the reading of these files is consistent with what is done in the batch predictor and pyfunc server `setup.py` files.

*I've selected `1.26.1` because it's the latest version of Mlflow available that is before Mlflow v2, that still allows `pandas==2.2.0` to be installed (this is a hard requirement needed by the `arize` package).

# Modifications
- Update Mlflow version to `1.26.1` (and other dependencies) in the SDK as well as observation publisher's requirements file since the unit tests there import the SDK's requirements file's dependencies
- ~~Update unit tests to ensure that the updated `PythonModelContext` from Mlflow is initialised with the required arguments~~ (will only be needed for newer versions of Mlflow)

# Tests

# Checklist
- [x] Added PR label
- [ ] Added unit test, integration, and/or e2e tests
- [x] Tested locally
- [ ] Updated documentation
- [ ] Update Swagger spec if the PR introduce API changes
- [ ] Regenerated Golang and Python client if the PR introduces API changes

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: python/sdk/setup.py
Comment (deadlycoconuts) [inline python/sdk/setup.py:26]: Refactored the listing of requirements away into a separate file, just like how it's done in the batch predictor and pyfunc server packages (see https://github.com/caraml-dev/merlin/blob/main/python/batch-predictor/setup.py#L24 for an example).

**Review by mbruner — APPROVED** [review]
lgtm, thanks!

**Comment by deadlycoconuts** [comment]
Thanks for the quick review. I've lowered the upgrade back to 1.26.1 because it's latest version (below version 2) that does not have any restrictions preventing `pandas==2.2.0` from being installed (which is a requirement of the `arize` package we're using). 

I'll let the e2e tests pass and merge this after that.

**Pull request merged**
